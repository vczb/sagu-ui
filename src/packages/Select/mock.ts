export const mock = ['New York', 'Los Angeles', 'Chicago', 'Houston']
