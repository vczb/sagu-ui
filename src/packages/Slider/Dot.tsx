import React, { BaseHTMLAttributes } from 'react'
import * as S from './styles'

export type DotType = {
  onClick: (index: number) => void
  active: boolean
} & BaseHTMLAttributes<HTMLDivElement>

const Dot = ({ onClick, active }: DotType) => {
  return <S.Dot data-testid={'dot'} onClick={onClick} active={active} />
}

export default Dot
