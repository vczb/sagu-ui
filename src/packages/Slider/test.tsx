import React from 'react'
import { render, screen, fireEvent } from '../../utils/testUtils'
import Slider from './'
import { mock } from './mock'

describe('<Slider />', () => {
  it('should render correctly', () => {
    const { container } = render(<Slider images={mock} />)

    expect(container.firstChild).toBeDefined()
  })

  it('should display all images', () => {
    const { container } = render(<Slider images={mock} />)

    // +2 because of the buttons
    expect(container.firstChild?.childNodes.length).toBe(mock.length + 2)
  })

  describe('Buttons', () => {
    it('should not appear if there are no images', () => {
      const { container } = render(<Slider images={[]} />)

      expect(container.firstChild?.childNodes.length).toBe(0)
    })

    it('should not appear if there is only one image', () => {
      const { container } = render(<Slider images={[mock[0]]} />)

      expect(container.firstChild?.childNodes.length).toBe(1)
    })
    it('should be at the correct stage at the beginning', () => {
      const { container } = render(<Slider images={mock} />)

      const [prevButton, nextButton] = Array.from(
        container.querySelectorAll('button')
      )

      expect(prevButton).toHaveAttribute('disabled')
      expect(nextButton).not.toHaveAttribute('disabled')
    })

    it('should be at the correct stage if another image is selected', () => {
      const { container } = render(<Slider images={mock} />)

      const [prevButton, nextButton] = Array.from(
        container.querySelectorAll('button')
      )

      nextButton.click()

      expect(prevButton).not.toHaveAttribute('disabled')
      expect(nextButton).not.toHaveAttribute('disabled')
    })

    it('should be at the correct stage if the last image is selected', () => {
      const { container } = render(<Slider images={mock.slice(0, 3)} />)

      const [prevButton, nextButton] = Array.from(
        container.querySelectorAll('button')
      )

      nextButton.click()
      nextButton.click()

      expect(prevButton).not.toHaveAttribute('disabled')
      expect(nextButton).toHaveAttribute('disabled')
    })

    it('should be able to change the images by clicking on the dots', () => {
      render(<Slider images={mock} dots />)
      const [prevButton, nextButton] = screen.getAllByRole('button')
      expect(prevButton).toHaveAttribute('disabled')
      expect(nextButton).not.toHaveAttribute('disabled')

      const dots = screen.getAllByTestId('dot')
      const dotForLastImage = dots[9]
      fireEvent.click(dotForLastImage)

      expect(prevButton).not.toHaveAttribute('disabled')
      expect(nextButton).toHaveAttribute('disabled')

      fireEvent.click(prevButton)

      expect(prevButton).not.toHaveAttribute('disabled')
      expect(nextButton).not.toHaveAttribute('disabled')
    })

    it('should render loading placeholder if loading is set to true', () => {
      const { container } = render(<Slider images={mock} loading />)

      expect(screen.queryAllByRole('img')).toHaveLength(0)
      expect(container).toMatchSnapshot()
    })
  })
})
