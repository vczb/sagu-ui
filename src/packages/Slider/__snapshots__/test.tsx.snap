// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Buttons should render loading placeholder if loading is set to true 1`] = `
.c0 {
  max-width: 40rem;
  height: 20rem;
  color: #EFF0F9;
  background-image: linear-gradient( to right, currentColor 0%, #E6E6E6 20%, currentColor 40%, currentColor 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholder 1s linear infinite forwards;
  animation: placeholder 1s linear infinite forwards;
}

<div>
  <div
    class="c0"
  />
</div>
`;
